File: Java/src/com/dbc/algorithm/Array/NonDecreasingArray.java
package com.dbc.algorithm.Array;

public class NonDecreasingArray {
    public static boolean checkPossibility(int[] nums) {
        int length = nums.length;
        int count = 0;
        for (int i = 0; i < length - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                count++;
                if (count > 1) {
                    return false;
                }
                if (i > 0 && nums[i - 1] > nums[i + 1]) {
                    nums[i + 1] = nums[i];
                }
            }
        }
        return true;
    }
}

File: core/tmp/Python/array/non_decreasing_array.py
from typing import List

def checkPossibility(nums: List[int]) -> bool:
    n = len(nums)
    count = 0
    for i in range(n - 1):
        x, y = nums[i], nums[i + 1]
        if x > y:
            count += 1
            if count > 1:
                return False
            if i > 0 and y < nums[i - 1]:
                nums[i + 1] = x

    return True

if __name__ == "__main__":
    print(checkPossibility([4,2,3]))

